Patch-List-Repo/src/main/java/org/wso2/patchvalidator/service/Signer.java

```diff
@@ -31,82 +34,108 @@ static StringBuilder sign() {
-            System.out.println("Patch List is :" + patchesList);
+            patchesList = UmtClient.getPatchList("ReadyToSign");
+            LOG.info("------------------------------------------------------------------");
+            LOG.info("                    Patch list from UMT Database                  ");
+            LOG.info("------------------------------------------------------------------");
+            for (String patch : patchesList){
+                LOG.info(patch);
+            }
         } catch (Exception ex) {
-            System.out.println("Exception occurred when searching patches in the governance registry" +  " Exception: "+ ex);
+            LOG.info("Exception occurred when searching patches in the governance registry" +
+                    " Exception: "+ ex);
         }
 
         if (patchesList.size() < 1) {
-            System.out.println("No patches in the \"Ready to sign\" state");
-        }
+            LOG.info("No patches in the \"Ready to sign\" state");
+        } else {
+            /**
+             * iterate the patches list and check the state in registry
+             */
+
+            for (String patch : patchesList) {
+                //get version and patch id from patch name string
+                try {
+                    //WSO2-CARBON-PATCH-4.4.0-2912
+                    String[] patchReplacedNameArr = patch.replace("WSO2-CARBON-PATCH-", "")
+                            .split("-");
+                    carbonVersion = patchReplacedNameArr[0].trim();//carbon version - 4.2.0/4.4.0/5.0.0
+                    patchId = patchReplacedNameArr[1].trim(); //2912
+                } catch (Exception ex) {
+                    LOG.info("Patch name retrieved from greg not in the correct format, patch:" + patch + " Exception: "+ ex);
+                    continue;
+                }
+
+                //get PMT patch JSON
+                try {
+                    patchJson = PmtClient.getPatchInfo(carbonVersion, patchId);
+                    pmtResultArr = (JSONArray) patchJson.get("pmtResult");
+                } catch (ServiceException ex) {
+                    LOG.info("Retrieving patch json failed, patch:" + patch+ " Exception: "+ ex);
+                    continue;
+                }
+
+                try {
+                    patchInfo = new PatchInfo(pmtResultArr);
+                } catch (Exception ex) {
+                    LOG.info("Creating object from pmt patch json failed, patch name: \"" + patch + "\". " +
+                            " Exception: "+ ex);
+                    continue;
+                }
+
+                if (patchInfo.getPatchLifeCycleState().equals("ReadyToSign")) {
+                    hMap.put(patch, patchInfo.getWumReleasedTimestamp());
+                } else {
+                    LOG.info("======================Unable proceed to sign======================");
+                    LOG.info("Patch: "+ patch+ " State : "+patchInfo.getPatchLifeCycleState());
```

Comment:
```suggestion
                    LOG.info("Patch: " + patch + " State : " + patchInfo.getPatchLifeCycleState());
```
Please format these lines when you append strings using `+`

Reply:
done